Src/DependencyCollector/Shared/Implementation/EventHandlers/ServiceBusDiagnosticsEventHandler.cs

```diff
@@ -0,0 +1,78 @@
+﻿namespace Microsoft.ApplicationInsights.DependencyCollector.Implementation.EventHandlers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Globalization;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility;
+
+    /// <summary>
+    /// Implements ServiceBus DiagnosticSource events handling.
+    /// </summary>
+    internal class ServiceBusDiagnosticsEventHandler : DiagnosticsEventHandlerBase
+    {
+        public const string DiagnosticSourceName = "Microsoft.Azure.ServiceBus";
+        private const string EntityPropertyName = "Entity";
+        private const string EndpointPropertyName = "Endpoint";
+
+        internal ServiceBusDiagnosticsEventHandler(TelemetryConfiguration configuration) : base(configuration)
+        {
+        }
+
+        public override void OnEvent(KeyValuePair<string, object> evnt, DiagnosticListener ignored)
+        {
+            Activity currentActivity = Activity.Current;
+
+            switch (evnt.Key)
+            {
+                case "Microsoft.Azure.ServiceBus.ProcessSession.Stop":
+                case "Microsoft.Azure.ServiceBus.Process.Stop":
+                    this.OnRequest(evnt.Key, evnt.Value, currentActivity);
+                    break;
+                case "Microsoft.Azure.ServiceBus.Exception":
+                    break;
+                default:
```

Comment:
This seems to be catch-all to generate dependency from every event other than 3 above. Are we sure that any new events that may appear in `ServiceBus` namespace are "dependency-worthy"?

Reply:
This is a philosophical question :) If new instrumentation is added to EH/SB SDK should we ignore it or collect as something? We think the latter is a better solution as users will start seeing additional telemetry without a need to upgrade AI SDK. If that telemetry is not desired by them they still have an option to ignore it through config (```IncludeDiagnosticSourceActivities``` whitelist)

Reply:
That being said I did change the logic for EH listener to explicitly list the dependency activities (*Send* and *Receive*) as we are currently adding new instrumentation there which will produce *Process* activity which will be collected as request. Once that change is approved (https://github.com/Azure/azure-event-hubs-dotnet/pull/214) we will add support for handling it in AI SDK and update this logic to catch-all

Reply:
RE: "We think the latter is a better solution as users will start seeing additional telemetry without a need to upgrade AI SDK."
Agree, it's a bit philosophical; however, I would expect us to limit this philosophy to auto-accept only a certain patterns of the events to make them Dependency Telemetry Items, e.g. Start/Stop pattern or similar. Converting all events from the code we do not own exposes us to the risk of sending telemetry for every light operation someone decided to mark on the other side. That leads to billing implications and will be most likely corrected by the customer only after the first bill. "Converting pattern" obviously has similar risks but it's a bit more contained.

Reply:
So yes - we are only collecting Activities based on Start/Stop/Exception event naming convention. That being said the inheritance hierarchy does not make it clear what kinds of event names would get here and changes in base classes could affect this behavior so I've added an explicit check for ```.Stop``` suffix in catch-all block for SB (EH has no catch-all at this point)